modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/resources/manifest/ExportedArtifactsCollector.java

```diff
@@ -162,18 +161,28 @@ public void visitArrayType(ArrayType arrayType) {
 
       @Override
       public void visitObject(ObjectType objectType) {
-        clazz.set(getType(objectType));
+        Optional<ClassInformationAnnotation> classInformation = objectType.getAnnotation(ClassInformationAnnotation.class);
+        if (classInformation.isPresent() && !classInformation.get().getGenericTypes().isEmpty()) {
+          classInformation.get().getGenericTypes().forEach(generic -> exportedClasses.add(getClassForName(generic)));
+        }
+        exportedClasses.add(getType(objectType));
       }
 
       @Override
       public void visitString(StringType stringType) {
         Optional<EnumAnnotation> enumAnnotation = stringType.getAnnotation(EnumAnnotation.class);
         if (enumAnnotation.isPresent()) {
-          clazz.set(getType(stringType));
+          exportedClasses.add(getType(stringType));
         }
       }
-    });
 
-    return Optional.ofNullable(clazz.get());
+      private Class<?> getClassForName(String name) {
```

Comment:
use `org.mule.runtime.core.util.ClassUtils#loadClass(java.lang.String, java.lang.ClassLoader)` instead. The class loader used should be the TCCL